### PR TITLE
Animation Fix and Revert back to my Custom NavigationStack

### PR DESCRIPTION
**Pull request by amirsaam**

### CODE REVIEW

File: PlayCover/Utils/NavigationStack.swift [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    case none, defaultTranisition, custom(AnyTransition)
+
+    var id: String {
+        switch self {
+        case .none: return "none"
+        case .defaultTranisition: return "default"
+        case .custom: return "custom"
+        }
+    }
```
Comment (KhoraLee) [inline PlayCover/Utils/NavigationStack.swift:23]:
```suggestion
    static func == (lhs: StackNavigationTransition, rhs: StackNavigationTransition) -> Bool {
        switch (lhs, rhs) {
        case (.none, .none):
            return true
        case (.defaultTranisition, .defaultTranisition):
            return true
        case (.custom, .custom):
            return true
        default:
            return false
        }
    }

    case none, defaultTranisition, custom(AnyTransition)
```

File: PlayCover/Utils/NavigationStack.swift [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    init(currentSubview: Binding<AnyView>,
+         showingSubview: Binding<Bool>,
+         transition: StackNavigationTransition,
+         @ViewBuilder rootView: @escaping () -> RootContent) {
+        self._currentSubview = currentSubview
+        self._showingSubview = showingSubview
+        self.transition = transition
+        self.rootView = rootView
+    }
```
Comment (KhoraLee) [inline PlayCover/Utils/NavigationStack.swift:54]: IIRC, `@escaping` is not good to use in SwiftUI. And making transition `@State` variable doesn't seem necessary.
```suggestion
    let rootView: RootContent
    let transition: StackNavigationTransition

    @Binding var currentSubview: AnyView
    @Binding var showingSubview: Bool

    init(currentSubview: Binding<AnyView>,
         showingSubview: Binding<Bool>,
         transition: StackNavigationTransition,
         @ViewBuilder rootView: () -> RootContent
    ) {
        self.rootView = rootView()
        self.transition = transition
        self._currentSubview = currentSubview
        self._showingSubview = showingSubview
    }
```

File: PlayCover/Utils/NavigationStack.swift [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                                ? .move(edge: .leading)
+                                : transition.anyTransition)
+            } else {
+                StackNavigationSubview(currentSubview: $currentSubview,
+                                       isVisible: $showingSubview,
+                                       transition: transition) {
+                    currentSubview
+                        .transition(transition.anyTransition)
+                        .zIndex(1)
+                }
```
Comment (KhoraLee) [inline PlayCover/Utils/NavigationStack.swift:71]: Is there any reason declaring and using StackNavigationSubview?

File: PlayCover/Utils/NavigationStack.swift (hunk 1 above)
Comment (amirsaam) [reply PlayCover/Utils/NavigationStack.swift:23]: Done!

File: PlayCover/Utils/NavigationStack.swift (hunk 2 above)
Comment (amirsaam) [reply PlayCover/Utils/NavigationStack.swift:54]: Thanks!

File: PlayCover/Utils/NavigationStack.swift (hunk 3 above)
Comment (amirsaam) [reply PlayCover/Utils/NavigationStack.swift:71]: Actually I don't remember why I did it but now I see it's redundant as you said. Thanks for mentioning it!

**Pull request merged**